pincer/commands/commands.py

```diff
@@ -705,6 +739,133 @@ async def add_commands(self, commands: List[AppCommand]):
+                # to the subcommand group that we want to modify to hold `cmd`
+
+                for cmd_in_children in children:
+                    if (
+                        cmd_in_children.name == sub_command_group.name
+                        and cmd_in_children.description == sub_command_group.description
+                        and cmd_in_children.type == sub_command_group.type
+                    ):
+                        sub_command_group = cmd_in_children
+                        break
+                else:
+                    children.append(sub_command_group)
+
+                sub_command_group.options.append(AppCommandOption(
+                    name=cmd.app.name,
+                    description=cmd.app.description,
+                    type=AppCommandOptionType.SUB_COMMAND,
+                    options=cmd.app.options,
+                ))
+
+                continue
+
+            if cmd.group:
+                # Any command at this point will only have one level of nesting.
+                #
+                # Command
+                #    subcommand
+                #
+                # A subcommand object is what is being generated here. If there is no
+                # top level command, it will be created here.
+
+                # `key` represents the hash value for the top-level command that will
+                # hold the subcommand.
+
+                key = _hash_app_command_params(
+                    cmd.group.name,
+                    cmd.app.guild_id,
+                    AppCommandOptionType.SUB_COMMAND,
+                    None,
+                    None
+                )
+
+                if key not in ChatCommandHandler.built_register:
+                    ChatCommandHandler.built_register[key] = AppCommand(
+                        name=cmd.group.name,
+                        description=cmd.group.description,
+                        type=AppCommandOptionType.SUB_COMMAND,
+                        guild_id=cmd.app.guild_id,
+                        options=[]
+                    )
+
+                # No checking has to be done before appending `cmd` since it is the
+                # lowest level.
+                ChatCommandHandler.built_register[key].options.append(
+                    AppCommandOption(
+                        name=cmd.app.name,
+                        description=cmd.app.description,
+                        type=AppCommandType.CHAT_INPUT,
+                        options=cmd.app.options
+                    )
+                )
+
+                continue
+
+            # All single-level commands are registered here.
+            ChatCommandHandler.built_register[
+                _hash_app_command(cmd.app, cmd.group, cmd.sub_group)
+            ] = cmd.app
+
+    def get_local_registered_commands(self):
+        return [
+            registered_cmd for registered_cmd
+            in ChatCommandHandler.built_register.values()
+        ]
```

Comment:
I'm not sure about the point to transform values to a list as this shouldn't be a object you modify, so i would just return the value i guess.

```suggestion
        return ChatCommandHandler.built_register.values()
```

or shouldn't it be
```suggestion
        return [
            registered_cmd.app for registered_cmd
            in ChatCommandHandler.built_register.values()
        ]
```
or maybe even
```suggestion
        return tuple(
            registered_cmd.app for registered_cmd
            in ChatCommandHandler.built_register.values()
        )
```

Reply:
The first one is right. Kinda slipped my mind when i was changing it xD